website/docs/r/lb_service_monitor.html.markdown

```diff
@@ -0,0 +1,95 @@
+---
+layout: "vcd"
+page_title: "vCloudDirector: vcd_lb_service_monitor"
+sidebar_current: "docs-vcd-resource-lb-service-monitor"
+description: |-
+  Provides a NSX load balancer service monitor resource.
+---
+
+# vcd\_lb\_service\_monitor
+
+Provides a vCloud Director Edge Gateway Load Balancer Service Monitor resource. A service monitor 
+defines health check parameters for a particular type of network traffic. It can be associated with
+a pool. Pool members are monitored according to the service monitor parameters. 
+
+~> **Note:** To make load balancing work one must ensure that load balancing is enabled on edge gateway. This depends 
+on NSX version to work properly. Please refer to [VMware Product Interoperability Matrices](https://www.vmware.com/resources/compatibility/sim/interop_matrix.php#interop&29=&93=) 
+to check supported vCloud director and NSX for vSphere configurations.
+
+~> **Note:** The vCloud Director API for NSX supports a subset of the operations and objects defined in the NSX vSphere 
+API Guide. The API supports NSX 6.2, 6.3, and 6.4.
+
+Supported in provider *v2.4+*
+
+## Example Usage
+
+```hcl
+provider "vcd" {
+  user     = "${var.admin_user}"
+  password = "${var.admin_password}"
+  org      = "System"
+  url      = "https://AcmeVcd/api"
+}
+
+resource "vcd_lb_service_monitor" "monitor" {
+  org          = "my-org"
+  vdc          = "my-org-vdc"
+  edge_gateway = "my-edge-gw"
+
+  name        = "testing-monitor-yy"
+  interval    = "5"
+  timeout     = "20"
+  max_retries = "3"
+  type        = "http"
+  http_method = "GET"
+  url         = "/health"
+  extension = {
+    content-type = "application/json"
+    linespan     = ""
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `edge_gateway` - (Required) The name of the edge gateway on which the service monitor is to be created
+* `vdc` - (Optional) The name of VDC to use, optional if defined at provider level
+* `org` - (Optional) The name of organization to use, optional if defined at provider level. Useful when connected as sysadmin working across different organisations
+* `name` - (Required) Service Monitor name
+
+* `interval` - (Required) Interval at which a server is to be monitored using the specified Method. 
+* `timeout` - (Required) Maximum time in seconds within which a response from the server must be received 
+* `max_retries` - (Required) Number of times the specified monitoring Method must fail sequentially before the server is declared down
+* `type` - (Required) Select the way in which you want to send the health check request to the server — `http`, `https`, 
+`tcp`, `icmp`, or `udp`. Depending on the type selected, the remaining attributes are allowed or not
```

Comment:
Is this the field, if specified incorrectly, we return an error from NSX directly and NSX shows these values in upper case? How can we make sure the user knows to use lower case instead?

Reply:
It is a bit of dumb situation inside the API. API accepts both cases for `type` and `method`, but returns in specific cases. The `type` is always being returned in lower case (doesn't matter in which case you submit it) while the `method` always returns uppercase. I didn't to allow confusion about settings state, casing.

My idea is that I have put in terraform validators on these fields to verify case (`type` and `method`) so that you are unable to submit incorrectly cased string.